src/components/layout/RegisterNameForm/RegisterNameForm.tsx

```diff
@@ -0,0 +1,135 @@
+import { useState } from 'react';
+
+import { useRegistrationState } from '../../../state/contexts/RegistrationState';
+import { ArweaveTransactionId } from '../../../types';
+import { isArweaveTransactionID } from '../../../utils/searchUtils';
+import Dropdown from '../../inputs/Dropdown/Dropdown';
+import UpgradeTier from '../UpgradeTier/UpgradeTier';
+import './styles.css';
+
+function RegisterNameForm() {
+  const [antFlow, setAntFlow] = useState(
+    'Create an Arweave Name Token (ANT) for me',
+  );
+  const [antAddress, setAntAddress] = useState<ArweaveTransactionId>();
+  const [{ domain, ttl, targetID }, dispatchRegisterState] =
+    useRegistrationState();
+
+  function handleTargetID(e: any) {
+    const id = e.target.value.trim();
+    // todo: add some more advanced error handling and notifications to check if id is a dataID or a value transfer
+    dispatchRegisterState({
+      type: 'setTargetID',
+      payload: id,
+    });
+    return;
+  }
+
+  return (
+    <>
+      <div className="register-name-modal center">
+        <div className="section-header">{domain} is available!</div>
+        <div className="section-header">Register Domain</div>
+        <div className="register-inputs center">
+          <div className="input-group center column">
+            <Dropdown
+              showSelected={true}
+              showChevron={true}
+              selected={antFlow}
+              setSelected={setAntFlow}
+              options={{
+                create: 'Create an Arweave Name Token (ANT) for me',
+                useExisting: 'I have an Arweave Name Token (ANT) I want to use',
+              }}
+            />
+            {/**TODO: add global error state to mark issues with the target id */}
+            {antFlow === 'Create an Arweave Name Token (ANT) for me' ? (
+              <input
+                className="data-input center"
+                placeholder="Target ID"
+                value={targetID}
+                onChange={(e) => handleTargetID(e)}
+                maxLength={43}
+                style={
+                  targetID && isArweaveTransactionID(targetID)
+                    ? { border: '2px solid var(--success-green)' }
+                    : targetID && !isArweaveTransactionID(targetID)
+                    ? { border: '2px solid var(--error-red)' }
+                    : {}
+                }
```

Comment:
<img width="1743" alt="image" src="https://user-images.githubusercontent.com/12800001/208141832-d7127d81-cd31-4c0d-a2fe-4b58edde85c8.png">


also, the tx id for a current ANT fails the validation `gh673M0Koh941OIITVXl9hKabRaYWABQUedZxW-swIA`